src/thread_task.rs

```diff
@@ -679,37 +701,42 @@ unsafe fn abort_frame(f: *mut Rav1dFrameContext, error: Rav1dResult) {
         FRAME_ERROR,
     );
     rav1d_decode_frame_exit(&mut *f, error);
-    pthread_cond_signal(&mut (*f).task_thread.cond);
+    (*f).task_thread.cond.notify_one();
 }
 
 #[inline]
-unsafe fn delayed_fg_task(c: *const Rav1dContext, ttd: *mut TaskThreadData) {
-    let in_0 = (*ttd).delayed_fg.in_0;
-    let out = (*ttd).delayed_fg.out;
+unsafe fn delayed_fg_task(
+    c: *const Rav1dContext,
+    ttd: *mut TaskThreadData,
```

Comment:
Can these be made refs?  All the mutability is unchecked right now.  A `&mut TaskThreadData` can't exist alongside a `MutexGuard<TaskThreadData_inner>`.